privilege/privileges/cache.go

```diff
@@ -535,6 +536,14 @@ func (p *MySQLPrivilege) buildDBMap() {
 	for _, record := range p.DB {
 		dbMap[record.User] = append(dbMap[record.User], record)
 	}
+
+	// Sort the records to make the matching rule work.
+	for _, records := range dbMap {
+		sort.Slice(records, func(i, j int) bool {
```

Comment:
if you rewrite  ```compareBaseRecord```  as follows：
```
func compareBaseRecord(x, y baseRecord) bool
```

you can use ```slice.SortFunc``` to replace ```sort.Slice```.

```
slice.SortFunc(records, compareBaseRecord)
```

it is more simple and avoids BCE in the Less function.

Reply:
`compareBaseRecord` can't work.

```
cat aaa.go
package main

import (
	"golang.org/x/exp/slices"
)

type baseT struct {
	int
}

type Extend struct {
	baseT
}

func compareBaseT(x, y baseT) bool {
	return x.int < y.int
}

func main() {
	xx := []Extend{}
	slices.SortFunc(xx, compareBaseT)
}
```

```
go run aaa.go
# command-line-arguments
./aaa.go:21:22: type func(x baseT, y baseT) bool of compareBaseT does not match inferred type func(a Extend, b Extend) bool for func(a E, b E) bool
```

It need a `compareDBRecord` function